### PR TITLE
[ Custom ] Add custom optimizer example in Application

### PR DESCRIPTION
**Pull request by jijoongmoon**
This PR includes the custom optimizer 'momentum' example in Applicaiton/Custom

It adds the testcase and demo implementaiton

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: Applications/Custom/momentum.cpp [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+/**
+ * Copyright (C) 2021 Parichay Kapoor <pk.kapoor@samsung.com>
+ *
+ * @file   momentum.h
+ * @date   1 June 2021
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Parichay Kapoor <pk.kapoor@samsung.com>
+ * @bug    No known bugs except for NYI items
+ * @brief  This is the Momentum optimizer.
+ */
```
Comment (DonghakPark) [inline Applications/Custom/momentum.cpp:11]: I think Author information didn't updated

File: Applications/Custom/momentum.h [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+/**
+ * Copyright (C) 2021 Parichay Kapoor <pk.kapoor@samsung.com>
+ *
+ * @file   momentum.h
+ * @date   1 June 2021
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Parichay Kapoor <pk.kapoor@samsung.com>
+ * @bug    No known bugs except for NYI items
+ * @brief  This is the Momentum optimizer.
+ */
```
Comment (DonghakPark) [inline Applications/Custom/momentum.h:11]: ditto

**Review by DonghakPark — APPROVED** [review]
LGTM!

**Review by baek2sm — APPROVED** [review]
LGTM

File: Applications/Custom/momentum.cpp (hunk 1 above)
Comment (jijoongmoon) [reply Applications/Custom/momentum.cpp:12]: fixed. Thanks.

File: Applications/Custom/momentum.h (hunk 2 above)
Comment (jijoongmoon) [reply Applications/Custom/momentum.h:12]: fixed. Thanks.

**Pull request merged**